girder/models/item.py

```diff
@@ -540,3 +540,83 @@ def checkConsistency(self, stage, progress=None):
                     progress.update(increment=1, message='Checking items (%d '
                                     'left)' % itemsLeft)
             return numItems, itemsCorrected
+
+    def _mergeMetadataHelper(self, oldDoc, newDoc, type='item'):
+        # metadata
+        if 'meta' in oldDoc:
+            if 'meta' in newDoc:
+                newDoc['meta'] = dict(
+                    newDoc['meta'].items() + oldDoc['meta'].items())
+            else:
+                newDoc['meta'] = oldDoc['meta']
+        # other fields
+        filteredDoc = self.model(type).filter(oldDoc)
+        for key in oldDoc:
+            if key not in filteredDoc and key not in oldDoc:
+                newDoc[key] = copy.deepcopy(oldDoc[key])
```

Comment:
Not sure what this loop is intended to do, but it appears that it will always be a no-op:

```
for key in oldDoc:
   if ... and key not in oldDoc:
      <never reached>
```